src/app/shared/pipes/weight.pipe.ts

```diff
@@ -0,0 +1,20 @@
+import { Pipe, PipeTransform } from "@angular/core";
+
+import { Weight } from "@shared/models/types/weight.type";
+
+@Pipe({
+    name: "weightToGrams"
+})
+export class WeightPipe implements PipeTransform {
+    transform(weight: Weight): number {
+        const isGrams = !weight.includes("kg");
+        const rawWeightValue = weight.split(/kg|g/)[0];
+
+        let weightValue = Number.parseInt(rawWeightValue);
```

Comment:
here, we should use const. I get that you used let because you were reassigning the variable.

instead, we can do it in the return statement using the ternary operator (isKilos ? weightValue : weightValue / 1000)

The change is not necessarily to shorten the number of lines, but to avoid let as much as possible